app/services/customers/create_service.rb

```diff
@@ -32,7 +32,15 @@ def create_from_api(organization:, params:)
           return currency_result unless currency_result.success?
         end
 
-        customer.save!
+        ActiveRecord::Base.transaction do
+          customer.save!
+
+          if new_customer && params[:metadata]
+            params[:metadata].each { |m| create_metadata(customer:, args: m) }
+          elsif params[:metadata]
+            Customers::Metadata::UpdateService.new(customer:).call(params: params[:metadata])
```

Comment:
Any particular reason for this change? 🙂 `customer` is the variable that is used in all three methods in `Customers::Metadata::UpdateService` class so it might have sense to create `customer` instance variable in constructor...

Reply:
Yeah, customer and params will be instance variables.

It's just to be consistent with what we have in other services for update, [here](https://github.com/getlago/lago-api/blob/main/app/services/add_ons/update_service.rb#L5) for instance ;)

Reply:
In given example both parameters are passed to constructor. However, in your suggestion default constructor is used (and params are passed to the call method). I guess we should just change suggestion and follow the setup from given example? :)

Reply:
Look at this: https://github.com/getlago/lago-api/blob/main/app/services/base_service.rb#L135 :) 

That's the same for the given example: https://github.com/getlago/lago-api/blob/main/app/controllers/api/v1/add_ons_controller.rb#L24

Reply:
ahhh, I see:) Changed!